# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/nsFilePicker.h
# Commit: c3e51c48222d
# Full Hash: c3e51c48222d9078a517ddfd30dc3dd8e850acd6
# Author: Ray Kraesig <rkraesig@mozilla.com>
# Date: 2023-10-27 04:08:13
# Regressor Bug: 1837079
# File Overlap Count: 1
# Description:
#   Bug 1837079 - [7/10] Open Windows file picker out-of-process  r=gstoll,handyman,ipc-reviewers,nika,win-reviewers,mhowell
#   
#   When opening a new Windows file dialog, open it out-of-process if
#   possible. Fall back to opening it in-process if that fails. (This
#   behavior is configurable with a pref.)
# ==============================================================================

diff -r eb3288fb639f -r c3e51c48222d widget/windows/nsFilePicker.h
--- a/widget/windows/nsFilePicker.h	Thu Oct 26 18:21:28 2023 +0000
+++ b/widget/windows/nsFilePicker.h	Thu Oct 26 18:21:29 2023 +0000
@@ -83,6 +83,20 @@
   bool ShowFilePicker(const nsString& aInitialDir);
 
  private:
+  // Show the dialog (by default, remotely falling back to locally, or whatever
+  // is specified by the current config).
+  static Result<Maybe<Results>> ShowFilePickerImpl(
+      HWND aParent, FileDialogType type, nsTArray<Command> const& commands);
+  static Result<Maybe<nsString>> ShowFolderPickerImpl(
+      HWND aParent, nsTArray<Command> const& commands);
+
+  // Show the dialog out-of-process.
+  static Result<Maybe<Results>> ShowFilePickerRemote(
+      HWND aParent, FileDialogType type, nsTArray<Command> const& commands);
+  static Result<Maybe<nsString>> ShowFolderPickerRemote(
+      HWND aParent, nsTArray<Command> const& commands);
+
+  // Show the dialog in-process.
   static Result<Maybe<Results>> ShowFilePickerLocal(
       HWND aParent, FileDialogType type, nsTArray<Command> const& commands);
   static Result<Maybe<nsString>> ShowFolderPickerLocal(
